### PR TITLE
Right click delete files & folders

### PR DESCRIPTION
**Pull request by DivergentEuropeans**
- Added a delete option for files & folders in the right click menu
- Added a popup to confirm delete
- Added notification framework UI for delete responses

PR 1 of 2
PR 2 https://github.com/zowe/zlux-app-manager/pull/165

<Signed-off-by: Leanid Astrakou <lastrakou@rocketsoftware.com>

### CODE REVIEW

File: src/app/components/filebrowseruss/filebrowseruss.component.ts [hunk 1]
```diff
@@ -30,8 +30,10 @@ import { TreeNode } from 'primeng/primeng';
 import { Angular2InjectionTokens, Angular2PluginWindowActions } from 'pluginlib/inject-resources';
 import 'rxjs/add/operator/toPromise';
 import { SearchHistoryService } from '../../services/searchHistoryService';
-import { MatDialog, MatDialogConfig } from '@angular/material';
+import { MatDialog, MatDialogConfig, MatSnackBar } from '@angular/material';
 import { FilePropertiesModal } from '../file-properties-modal/file-properties-modal.component';
+import { DeleteFileModal } from '../delete-file-modal/delete-file-modal.component';
+import { ZoweNotification } from '../../../../../../../../../zlux-platform/base/src/notification-manager/notification';
```
Comment (1000TurquoisePogs) [inline src/app/components/filebrowseruss/filebrowseruss.component.ts]: Not wrong but brittle, using an alias in tsconfig is better such as in here https://github.com/zowe/zlux-app-manager/blob/master/virtual-desktop/tsconfig.json#L38

File: src/app/components/filebrowseruss/filebrowseruss.component.ts [hunk 2]
```diff
@@ -534,14 +559,38 @@ export class FileBrowserUSSComponent implements OnInit, OnDestroy {//IFileBrowse
...
       resp => {
+        this.sendNotification('Editor', 'Deleted: ' + pathAndName);
         this.updateUss(this.path);
       },
-      error => this.errorMessage = <any>error
+      error => {
+        if (error.status == '403') { //Forbidden (TODO: Implement ZSS functionality to return a 403 for permissions problems)
+          this.sendNotification('Editor', 'Incorrect permissions to delete: ' + pathAndName);
+        } else if (error.status == '410') { //Gone (TODO: Implement ZSS functionality to return a 410 for a file/folder that has already been deleted)
+          this.sendNotification('Editor', 'Failed to delete: ' + pathAndName);
```
Comment (1000TurquoisePogs) [inline src/app/components/filebrowseruss/filebrowseruss.component.ts]: Think of another string to differentiate between these distinct error cases when you can reproduce them.

File: src/app/components/filebrowseruss/filebrowseruss.component.ts [hunk 3]
```diff
@@ -534,14 +559,38 @@ export class FileBrowserUSSComponent implements OnInit, OnDestroy {//IFileBrowse
...
+        } else if (error.status == '500') { //Internal Server Error
+          this.sendNotification('Editor', 'Failed to delete: ' + pathAndName);
+        } else {
+          this.sendNotification('Editor', 'Uknown error occured for: ' + pathAndName);
```
Comment (1000TurquoisePogs) [inline src/app/components/filebrowseruss/filebrowseruss.component.ts]: If you dont know the error, at least show them the status code so they can figure it out.

File: src/app/components/filebrowseruss/filebrowseruss.component.ts [hunk 4]
```diff
@@ -534,14 +559,38 @@ export class FileBrowserUSSComponent implements OnInit, OnDestroy {//IFileBrowse
...
+      if (deleteSubscription.closed == false) {
+        this.snackBar.open('Deleting ' + pathAndName + '... Larger payloads may take longer. Please be patient.', 
+          'Dismiss', { duration: 5000,   panelClass: 'center' });
+      }
+    }, 4000);
   }
+
+  sendNotification(title: string, message: string): number {
+    let pluginId = this.pluginDefinition.getBasePlugin().getIdentifier();
+    let notification = new ZoweNotification(title, message, 1, pluginId, "org_zowe_zlux_editor_snackbar");
```
Comment (1000TurquoisePogs) [inline src/app/components/filebrowseruss/filebrowseruss.component.ts]: Please don't use a theme, but perhaps comment the use of it out, so we can show people how they might use a theme in the future.

File: src/app/components/filebrowseruss/filebrowseruss.component.ts (hunk 1 above)
Comment (DivergentEuropeans) [reply src/app/components/filebrowseruss/filebrowseruss.component.ts]: Unfortunately, was not able to get this to work. Tried all sorts of working with tsconfig and no dice

**Pull request merged**